code/modules/mob/living/carbon/give.dm

```diff
@@ -1,52 +1,167 @@
-/mob/living/carbon/verb/give(mob/living/carbon/target in oview(1))
+/**
+ * Toggles the [/datum/click_intercept/give] on or off for the src mob.
+ */
+/mob/living/carbon/verb/toggle_give()
 	set category = "IC"
-	set name = "Give"
+	set name = "Give Item"
 
-	if(!iscarbon(target)) //something is bypassing the give arguments, no clue what, adding a sanity check JIC
-		to_chat(usr, "<span class='danger'>Wait a second... \the [target] HAS NO HANDS! AHH!</span>")//cheesy messages ftw
+	if(HAS_TRAIT(src, TRAIT_OFFERING_ITEM))
+		to_chat(src, "<span class='warning'>You're already offering an item to someone!</span>")
 		return
+	if(istype(client.click_intercept, /datum/click_intercept/give))
+		give_icon.icon_state = "act_give_off"
+		QDEL_NULL(client.click_intercept)
+		return
+	var/obj/item/I = get_active_hand()
+	if(!I)
+		to_chat(src, "<span class='warning'>You don't have anything in your hand to give!</span>")
+		return
+	if(I.flags & NODROP)
+		to_chat(src, "<span class='warning'>[I] is stuck to your hand, you can't give it away!</span>")
+		return
+	if(I.flags & ABSTRACT)
+		to_chat(src, "<span class='warning'>That's not exactly something you can give.</span>")
+		return
+
+	give_icon.icon_state = "act_give_on"
+	new /datum/click_intercept/give(client)
+
+/**
+ * # Give click intercept
+ *
+ * While a mob has this intercept, left clicking on a carbon mob will attempt to offer their currently held item to that mob.
+ */
+/datum/click_intercept/give
+
+/datum/click_intercept/give/New(client/C)
+	..()
+	holder.mouse_pointer_icon = file("icons/mouse_icons/give_item.dmi")
+	to_chat(holder, "<span class='info'>You can now left click on someone to give them your held item.</span>")
+	RegisterSignal(holder.mob.get_active_hand(), list(COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_DROPPED), .proc/cancel_give)
+	RegisterSignal(holder.mob, list(SIGNAL_ADDTRAIT(TRAIT_HANDS_BLOCKED), COMSIG_CARBON_SWAP_HANDS), .proc/cancel_give)
 
-	if(target.incapacitated() || usr.incapacitated() || target.client == null)
+
+/datum/click_intercept/give/Destroy(force = FALSE, silent = FALSE)
```

Comment:
`force` is unused

Reply:
Has to be there, since the parent has it.

Reply:
Oh, my bad; i checked the immediate `/datum/click_intercept/Destroy`, didn't found the arg, but didn't check the base `/datum/Destroy`